### PR TITLE
Add development dependencies to the gemspec

### PR DESCRIPTION
**Pull request by fractaledmind**
And remove the one from the Gemfile (while fixing an old name reference)

You can see this bug in action by running the following Ruby script:
```ruby
require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  gem "sqlite3"
  gem "litestack"
end
```

The error thrown will be (truncated and simplified):
```
cannot load such file -- rails/railtie (LoadError)
from .../gems/litestack-0.2.6/lib/litestack/railtie.rb:1:in `<top (required)>'
from .../gems/litestack-0.2.6/lib/litestack.rb:19:in `require_relative'
from .../gems/litestack-0.2.6/lib/litestack.rb:19:in `<top (required)>'
```

The issue is that the gem doesn't provide the `rails/railtie` dependency on its own and thus cannot be used outside of a context that does (which is basically just a Rails app). 

This limits testing and creating reproducible bug scripts like the one above.

### CODE REVIEW

File: litestack.gemspec [hunk 1]
```diff
@@ -36,4 +36,8 @@ Gem::Specification.new do |spec|
   spec.add_dependency "hanami-router"
   spec.add_dependency "tilt"
   spec.add_dependency "erubi"
+  
+  spec.add_development_dependency "rake"
+  spec.add_development_dependency "railties"
```
Comment (marcoroth) [inline litestack.gemspec:41]: I was also running into this, but I'm wondering if we really need to add `railties` as a dependency. 

I think we would just need to update the `if` condition here:
https://github.com/oldmoe/litestack/blob/8a061b849f2e3dd137f398fca0fd99a8ea5451c0/lib/litestack.rb#L19

To something like `if defined(Rails::Railtie)`

Comment (marcoroth) [reply litestack.gemspec:41]: I went ahead and proposed that here: https://github.com/oldmoe/litestack/pull/20

Comment (fractaledmind) [reply litestack.gemspec:41]: I like your solution for production, but am only adding railties as a dev dependency to make testing easier

Comment (marcoroth) [reply litestack.gemspec:41]: Oh, I see. That makes more sense

**Comment by fractaledmind** [comment]
@oldmoe: This has been updated

**Pull request merged**